src/NMS.AMQP/Message/Facade/INmsMessageFacade.cs

```diff
@@ -15,47 +15,28 @@
  * limitations under the License.
  */
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-using Apache.NMS;
 
-
-namespace Apache.NMS.AMQP.Message.Cloak
+namespace Apache.NMS.AMQP.Message.Facade
 {
-    /// <summary>
-    /// Provider specific Cloak Interface from provider implementation.
-    /// </summary>
-    interface IMessageCloak : IMessage
+    public interface INmsMessageFacade
     {
-        byte[] Content
-        {
-            get;
-            set;
-        }
-
-        bool IsBodyReadOnly { get; set; }
-
-        bool IsPropertiesReadOnly { get; set; }
-
-        bool IsReceived { get; }
-
-        IMessageCloak Copy();
-
-        object GetMessageAnnotation(string symbolKey);
-
-        void SetMessageAnnotation(string symbolKey, object value);
-
-        object GetDeliveryAnnotation(string symbolKey);
-
-        void SetDeliveryAnnotation(string symbolKey, object value);
-
+        NmsMessage AsMessage();
+        void ClearBody();
         int DeliveryCount { get; set; }
-
         int RedeliveryCount { get; set; }
-
-        MessageAcknowledgementHandler AckHandler { get; set; }
-        
+        void OnSend(TimeSpan producerTtl);
+        string NMSMessageId { get; set; }
+        IPrimitiveMap Properties { get; }
+        string NMSCorrelationID { get; set; }
+        IDestination NMSDestination { get; set; }
+        TimeSpan NMSTimeToLive { get; set; }
+        MsgDeliveryMode NMSDeliveryMode { get; set; }
+        MsgPriority NMSPriority { get; set; }
+        bool NMSRedelivered { get; set; }
+        IDestination NMSReplyTo { get; set; }
+        DateTime NMSTimestamp { get; set; }
+        string NMSType { get; set; }
+        DateTime Expiration { get; set; }
+        double JmsMsgType { get; }
```

Comment:
Why the type double for JmsMsgType? The qpid implementation uses a signed byte. I found using Helloworld the JMS msg type annotation is of type is double. This breaks the JMS to amqp bind mapping restriction Section 3.2.4.1 for sending messages where the msg type annotation must be of type byte (signed byte if you want to inter-operable with qpid jms though). 
I would suggest changing the type of the JmsMsgType to sbyte (and everywhere else). Unless there is a very good reason not to.
```suggestion
        sbyte JmsMsgType { get; }
```

Reply:
You are absolutely right. I will make necessary adjustments. 

Are you sure that using byte would break interoperability with qpid jms?

Reply:
We can make a .net to java check tomorrow on a test platform. Simplest way to be sure

Reply:
Before I added the comment. I took a look at the qpid code which does a cast a to signed "byte" type on the annotation value and I verified the UnsignedByte class from proton can not be cast to "byte". However I have not tried to receive a message sent from the amqp provider to a qpid jms app yet.

I'm definitely in favour for a .net to java check.

Reply:
Confirmed. Using bytes instead of sbytes breaks interoperability.